### PR TITLE
Sw invite users to organization

### PR DESCRIPTION
**Pull request by SWvheerden**
This is for the invite users. 
The email functionality is not yet in. I am only creating the PR so that the API calls can be created so we can start using them in the front end.

### CODE REVIEW

**Comment by SWvheerden** [comment]
This build will fail until if the bn-db one is not merged in

File: src/controllers/organization_invites.rs [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        return application::unauthorized();
+    };
+    let mut actual_new_invite = new_org_invite.into_inner();
+    let email = actual_new_invite.user_email.clone();
+    let org_invite = match NewOrganizationInvite::commit(&mut actual_new_invite, &*connection) {
+        Ok(u) => u,
+        Err(e) => return HttpResponse::from_error(ConvertToWebError::create_http_error(&e)),
+    };
+    let mut was_user_found = true;
+    //we only care to add the user id if we can find it via the email
```
Comment (mikethetike) [inline src/controllers/organization_invites.rs]: Unused variable?

File: src/controllers/organization_invites.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            Ok(_u) => was_user_found = false,
+            Err(_e2) => was_user_found = false,
+        },
+        Err(_e) => was_user_found = false,
+    };
+    //send email here
+    HttpResponse::Created().json(org_invite)
+}
+
+fn do_request(data: (State<AppState>, Json<Info>, AuthUser, i16)) -> HttpResponse {
```
Comment (mikethetike) [inline src/controllers/organization_invites.rs]: Is there a more descriptive name?

File: src/controllers/organization_invites.rs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    let (state, info, user, status) = data;
+    let connection = state.database.get_connection();
+    if !user.is_in_role(Roles::User) {
+        return application::unauthorized();
+    };
+    let info_struct = info.into_inner();
+    let invite_details =
+        match OrganizationInvite::get_get_invite_details(&info_struct.token, &*connection) {
```
Comment (mikethetike) [inline src/controllers/organization_invites.rs]: `get_get`?

File: src/controllers/organization_invites.rs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            Ok(u) => u,
+            Err(e) => return HttpResponse::from_error(ConvertToWebError::create_http_error(&e)),
+        };
+    // see if we can stop non-intended user using invite.
+    //if this is a new user we cant do this check
+    if (invite_details.user_id != None) && (invite_details.user_id.unwrap() != info_struct.user) {
```
Comment (mikethetike) [inline src/controllers/organization_invites.rs]: Maybe rename `info_struct.user` to `info_struct.user_id`

File: src/routing.rs [hunk 5]
```diff
@@ -90,6 +90,20 @@ pub fn routes(app: &mut CorsBuilder<AppState>) -> App<AppState> {
         .resource("/auth/token/refresh", |r| {
             r.method(Method::POST).with(auth::token_refresh)
         })
+        .resource("/invite_user", |r| {
```
Comment (mikethetike) [inline src/routing.rs]: Add the `organizations/` prefix

File: src/routing.rs [hunk 6]
```diff
@@ -90,6 +90,20 @@ pub fn routes(app: &mut CorsBuilder<AppState>) -> App<AppState> {
...
+            r.middleware(AuthMiddleware::new());
+            r.method(Method::POST).with(organization_invites::create);
+        })
+        .resource("/accept_invite", |r| {
```
Comment (mikethetike) [inline src/routing.rs]: Add the `organizations/` prefix

File: src/routing.rs [hunk 7]
```diff
@@ -90,6 +90,20 @@ pub fn routes(app: &mut CorsBuilder<AppState>) -> App<AppState> {
...
+            r.middleware(AuthMiddleware::new());
+            r.method(Method::POST)
+                .with(organization_invites::accept_request);
+        })
+        .resource("/decline_invite", |r| {
```
Comment (mikethetike) [inline src/routing.rs]: Add the `organizations/` prefix

File: src/controllers/organization_invites.rs (hunk 1 above)
Comment (SWvheerden) [reply src/controllers/organization_invites.rs]: will be used sortly

**Pull request merged**